features/karaf/esh-core/src/main/feature/feature.xml

```diff
@@ -359,6 +384,23 @@
     <bundle>mvn:org.eclipse.smarthome.model/org.eclipse.smarthome.model.thing.runtime/${project.version}</bundle>
   </feature>
 
+  <feature name="esh-model-thing-ide" version="${project.version}">
+    <feature dependency="true">esh-model-thing</feature>
+    <bundle>mvn:org.eclipse.smarthome.model/org.eclipse.smarthome.model.thing.ide/${project.version}</bundle>
+  </feature>
+
+  <feature name="esh-model-lsp" version="${project.version}">
+    <requirement>esh.tp;filter:="(&amp;(feature=xtext-ide)(version&gt;=2.12.0)(!(version&gt;=2.13.0)))"</requirement>
+    <feature dependency="true">esh-tp-lsp4j</feature>
```

Comment:
Rely on esh-tp features using "requirement" and a filter expression as you have done in the line above for `xtext-ide` (first add your lsp4j optional feature dependency to the tp).

Reply:
awesome, I coudln't get the "requirement" statement to work - it seems like your above hint was the missing piece. THANKS! 
Will do so, of course...

Reply:
Some background: Every solution could create its own target platform feature definition.
The esh-base feature depends on the esh-tp feature.
All features that are part of esh-tp are using `dependency="true"` so they are only considered if someone depends on it.
We are using the requirement declaration so we state which esh targetplafrom feature is necessary...